core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -243,6 +243,14 @@ object UnitAutomation {
             wander(unit, stayInTerritory = true)
     }
 
+    /** @return true only if the unit has 0 movement left */
+    private fun tryAttacking(unit: MapUnit): Boolean {
+        for (numAttacks in unit.attacksThisTurn until unit.maxAttacksPerTurn()) {
+            if (BattleHelper.tryAttackNearbyEnemy(unit)) return true
```

Comment:
This should return upon the very first attack, no?

Reply:
I tested with the AI playing as the Chinese since their UU has 2 attacks and they were able to attack twice per turn.